speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp

```diff
@@ -51,3 +51,44 @@ TEST_CASE(MediatorTests, InterventionTestNoSignals, InterventionTest, (INTERVENT
 TEST_CASE(MediatorTests, InterventionTestOnlySignals, InterventionTest, (INTERVENTION_TYPE_ONLY_SIGNALS))
 TEST_CASE(MediatorTests, InterventionTestSharedControl, InterventionTest, (INTERVENTION_TYPE_SHARED_CONTROL))
 TEST_CASE(MediatorTests, InterventionTestCompleteTakeover, InterventionTest, (INTERVENTION_TYPE_COMPLETE_TAKEOVER))
+
+/// @brief            Writes a given mediator pointer to the Singletons/Mediator file
+/// @param p_mediator The mediator pointer to write to the file
+bool WriteMediator(SMediator* p_mediator)
+{
+    EXPECT_TRUE(SetupSingletonsFolder());
+    struct stat info
+    {
+    };
+    std::experimental::filesystem::path path = std::experimental::filesystem::temp_directory_path();
+    path.append("Singletons\\Mediator");
+    std::string pathstring = path.string();
+    const char* filepath = pathstring.c_str();
+    int err = stat(filepath, &info);
+    if (err == -1)
+    {
+        std::ofstream file(filepath);
+        file << p_mediator;
+        file.close();
+        return true;
+    }
+    return false;  // If we get here, SetupSingletonsFolder() already failed
+}
+
+TEST(MediatorTest, ReadFromFile)
+{
+    SMediator::GetInstance();
+    SMediator* oldMediator = SMediator::RemoveInstance();  // Makes sure we can't get into the if (m_instance) block
+    SMediator* fakeMediator = (SMediator*)malloc(sizeof(SMediator));
```

Comment:
please do not use malloc, but use new

Reply:
I now realise that this might be because mediator has a private constructor. I still think calling the private constructor is better.

Reply:
This was indeed because the constructor was private.
I now call that constructor in the same way I made the (new, and now removed) RemoveInstance() method available.